### PR TITLE
Ticket/989 missing validation of archive

### PR DESCRIPTION
**Pull request by Un1matr1x**
http://www.flying-bits.org/tracker.php?p=33&t=989

### CODE REVIEW

File: root/includes/functions_newspage.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+* Check if the $archive_var is a valid date for timestamp creation
+*/
+function newspage_check_archive_date($archive_var)
+{
+	if (preg_match("/[0-1][0-9]_[1-2][0-9]{3}/", $archive_var))
+	{
+		$archive = explode('_', $archive_var);
+		if ( (0 < $archive[0]) && ( 13 > $archive[0]) && (1970 <=  $archive[1]))
+		{
+			if ((is_int(2147483648)) || (2038 >  $archive[1]))
```
Comment (Un1matr1x) [inline root/includes/functions_newspage.php:28]: @nickvergessen 
this part need some testing on 64-Bit environment

**Comment by Un1matr1x** [comment]
This PR is at the moment untested, will do this later
some help from people with 64-bit environment needed for 1 test

File: root/includes/functions_newspage.php (hunk 1 above)
Comment (nickvergessen) [inline root/includes/functions_newspage.php:28]: I think `($archive[1] < 2038)` should be okay here, as it is only filtering existing posts. If someone posts topics now that are >  25 years into the future, he has to deal with the problems at other points first.

Comment (nickvergessen) [inline root/includes/functions_newspage.php:28]: Also please change the order of the checks from `constant ~ variable` to `variable ~ constant`

File: root/includes/functions_newspage.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+*/
+if (!defined('IN_PHPBB'))
+{
+	exit;
+}
+
+/**
+* Check if the $archive_var is a valid date for timestamp creation
+*/
+function newspage_check_archive_date($archive_var)
```
Comment (nickvergessen) [inline root/includes/functions_newspage.php:21]: Please rename to phpbb_newspage_validate_archive_date and move the function to the newspage.php

File: root/includes/functions_newspage.php (hunk 1 above)
Comment (Un1matr1x) [inline root/includes/functions_newspage.php:28]: so if we won't check for year > 2038 on 32-bit sys we can drop the other checks (except the regex) as well

?archive=12_2040 results in
0
0
AND topic_time >= 0 AND topic_time <= 0

?archive=15_1999
951868800
954547200
AND topic_time >= 951868800 AND topic_time <= 954547200

and this won't throw any error

so I'm going to close this PR and restart much smaller with a new branch

**Pull request closed without merging**